src/script/page/Modals/GroupCreation/GroupCreationModal.tsx

```diff
@@ -0,0 +1,405 @@
+
+        {stateIsPreferences && (
+          <>
+            <div className="modal-input-wrapper">
+              <TextInputForwarded
+                label={t('groupCreationPreferencesPlaceholder')}
+                placeholder={t('groupCreationPreferencesPlaceholder')}
+                uieName="enter-group-name"
+                name="enter-group-name"
+                errorUieName="error-group-name"
+                onCancel={() => setGroupName('')}
+                onChange={onGroupNameChange}
+                onBlur={event => {
+                  const {value} = event.target as HTMLInputElement;
+                  const trimmedName = value.trim();
+                  setGroupName(trimmedName);
+                }}
+                value={groupName}
+                isError={nameError.length > 0}
+                errorMessage={nameError}
+              />
+            </div>
+            {isTeam && (
+              <>
+                <div
+                  className="modal__info"
+                  style={{visibility: nameError.length > 0 ? 'hidden' : 'visible'}}
+                  data-uie-name="status-group-size-info"
+                >
+                  {t('groupSizeInfo', maxSize)}
+                </div>
+                <hr className="group-creation__modal__separator" />
+                <BaseToggle
+                  className="modal-style"
+                  isChecked={isGuestEnabled}
+                  setIsChecked={clickOnToggleGuestMode}
+                  extendedInfo
+                  extendedInfoText={t('guestRoomToggleInfoExtended')}
+                  infoText={t('guestRoomToggleInfo')}
+                  toggleName={t('guestOptionsTitle')}
+                  toggleId={'guests'}
+                />
+                <BaseToggle
+                  className="modal-style"
+                  isChecked={isServicesEnabled}
+                  setIsChecked={clickOnToggleServicesMode}
+                  extendedInfo
+                  extendedInfoText={t('servicesRoomToggleInfoExtended')}
+                  infoText={t('servicesRoomToggleInfo')}
+                  toggleName={t('servicesOptionsTitle')}
+                  toggleId="services"
+                />
+                <InfoToggle
+                  className="modal-style"
+                  dataUieName={'read-receipts'}
+                  info={t('readReceiptsToogleInfo')}
+                  isChecked={enableReadReceipts}
+                  setIsChecked={setEnableReadReceipts}
+                  isDisabled={false}
+                  name={t('readReceiptsToogleName')}
+                />
+              </>
+            )}
+          </>
+        )}
+      </ModalComponent>
+    </div>
+  );
+};
+
+export default {
+  GroupCreationModal,
+  init: (
+    conversationRepository: ConversationRepository,
+    searchRepository: SearchRepository,
+    teamRepository: TeamRepository,
+  ) => {
+    createRoot(document.getElementById('group-creation-modal-container')).render(
```

Comment:
Let's inject the dom element there and avoid coupling this generic component with some concrete dom instance. 

The `init` signature would then be 

```ts
init: (
   target: Element,
    conversationRepository: ConversationRepository,
    searchRepository: SearchRepository,
    teamRepository: TeamRepository,
  )
```

And the caller is responsible for injecting the right dom element. 

```ts
GroupCreationModal.init(document.getElementById('...'), conversationRepo,...)
```